### PR TITLE
tsp, support head boolean

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Enhancement for response error.

"*" is regarded as error, model with `@error` is regarded as error e.g. https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/foundations.tsp#L113-L124
Else be normal response (non-error response).

Most for the case of HEAD with 200+404 response. Detail https://teams.microsoft.com/l/message/19:906c1efbbec54dc8949ac736633e6bdf@thread.skype/1684272172008?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1684272172008&teamName=Azure%20SDK&channelName=TypeSpec%20(Cadl)%20Discussion%20%F0%9F%90%AE&createdTime=1684272172008

### CODE REVIEW

File: typespec-tests/src/main/java/com/cadl/response/ResponseServiceVersion.java
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/cadl/response/ResponseServiceVersion.java:1]: The rename here is caused by now ResponseClient also having api-version (previously on CoreClient has it), hence the name changed to the service name of ResponseServiceVersion

File: typespec-tests/tsp/response.tsp [hunk 1]
```diff
@@ -92,4 +106,22 @@ interface ResponseOp {
...
+    {},
+    {
+      @doc("A response containing headers related to the Job Schedule, if it exists.")
+      @statusCode
+      code: "200";
+    },
+    {
+      @doc("A response containing headers related to the Job Schedule, if it exists.")
+      @statusCode
+      code: "404";
```
Comment (haolingdong-msft) [inline typespec-tests/tsp/response.tsp:122]: what does default status code 404 mean?

Comment (weidongxu-microsoft) [reply typespec-tests/tsp/response.tsp:122]: that 404 is expected response (not an error)

swagger https://github.com/Azure/azure-rest-api-specs/blob/main/specification/batch/data-plane/Microsoft.Batch/stable/2022-10-01.16.0/BatchService.json#L2829-L2855

File: typespec-tests/tsp/response.tsp [hunk 2]
```diff
@@ -92,4 +106,22 @@ interface ResponseOp {
     @header("operation-location")
     operationLocation: ResourceLocation<Resource>;
   };
+
+  @route("/exists")
+  @head
```
Comment (haolingdong-msft) [inline typespec-tests/tsp/response.tsp:111]: Found the response model is boolean, will @head operation always return boolean response? is this handled by compiler or our emitter?

File: typespec-tests/tsp/response.tsp (hunk 1 above)
Comment (haolingdong-msft) [reply typespec-tests/tsp/response.tsp:122]: ok, I see, thanks.

File: typespec-tests/tsp/response.tsp (hunk 2 above)
Comment (weidongxu-microsoft) [reply typespec-tests/tsp/response.tsp:111]: No, not always. Batch got some other HEAD op that does not do this.

Comment (haolingdong-msft) [reply typespec-tests/tsp/response.tsp:111]: who will resolve response model type, compiler?

Comment (weidongxu-microsoft) [reply typespec-tests/tsp/response.tsp:111]: codegen. that part already exists https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/util/SchemaUtil.java#L99-L102

as well as azure-core

Comment (haolingdong-msft) [reply typespec-tests/tsp/response.tsp:111]: got it.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**